packages/bigquery/index.cjs

```diff
@@ -30,45 +31,66 @@ const standardizeResult = async (result) => {
 	return output;
 };
 
-const getCredentials = async (database) => {
-	try {
-		if (database) {
-			const creds = {
-				projectId: database.project_id,
-				credentials: {
-					client_email: database.client_email,
-					private_key: database.private_key
-				}
-			};
-			return creds;
-		} else {
-			const creds = {
-				projectId:
-					process.env['BIGQUERY_PROJECT_ID'] ||
-					process.env['project_id'] ||
-					process.env['PROJECT_ID'],
-				credentials: {
-					client_email:
-						process.env['BIGQUERY_CLIENT_EMAIL'] ||
-						process.env['client_email'] ||
-						process.env['CLIENT_EMAIL'],
-					private_key: (
-						process.env['BIGQUERY_PRIVATE_KEY'] ||
-						process.env['private_key'] ||
-						process.env['PRIVATE_KEY']
-					).replace(/\\n/g, '\n')
-				}
-			};
-			return creds;
-		}
-	} catch {
-		throw new Error('Missing database credentials');
+const getCredentials = (database = {}) => {
+	const authentication_method =
+		database.authenticator ??
+		process.env['BIGQUERY_AUTHENTICATOR'] ??
```

Comment:
This is going to have merge conflicts when the environment variables PR is merged.